rust/relay/src/server.rs

```diff
@@ -987,6 +1000,26 @@ where
     }
 }
 
+enum ResponseErrorLevel {
+    Warn,
+    Debug,
+}
+
+impl ResponseErrorLevel {
+    fn log(&self, method: Method, reason_phrase: &str) {
+        const TARGET: &str = "relay";
+        let log_string = format!("{method} failed: {reason_phrase}");
```

Comment:
I don't think this extra allocation when debug/warn is disabled is so bad? If that's the case we can duplicate it and inline it below.

Reply:
I think if you use `format_args`, it won't allocate?

Reply:
ah! yeah, forgot about that

Reply:
kind of hard to use it for this https://github.com/rust-lang/rust/issues/92698

Reply:
Boo